Correcting tests

daveomcd opened the pull request:
I took a shot at correcting the tests.  Here are the current results, I didn't finish the Active Record tests as I'm not knowledgable enough to see what next steps should be taken place there.

*Results from `bundle exec rake test ONLY_SQLSERVER=1`*

    Finished in 15.071858s, 15.5920 runs/s, 95.6750 assertions/s.
    235 runs, 1442 assertions, 0 failures, 0 errors, 1 skips

*Results from `bundle exec rake test ONLY_ACTIVERECORD=1`*

    Finished in 160.342181s, 32.8236 runs/s, 93.5562 assertions/s.
    5263 runs, 15001 assertions, 0 failures, 16 errors, 3 skips

lcreid commented on test/cases/coerced_tests.rb:
```diff
@@ -276,7 +276,7 @@ def test_add_column_without_limit_coerced
 module ActiveRecord
   class Migration
-    class ColumnsTest
+    class ColumnsTest < ActiveRecord::TestCase
```
Nice catch!

lcreid commented on test/cases/coerced_tests.rb:
```diff
@@ -759,6 +758,14 @@ def test_reverse_arel_assoc_order_with_function_coerced
     assert_equal topics(:second).title, topics.first.title
   end
+
+  coerce_tests! :test_pluck_with_from_includes_quoted_original_table_name
```
I'm curious why you're adding a new test along with fixing all the other tests?

daveomcd replied:
My mistake! This wasn't supposed to have been added.

> I'm curious why you're adding a new test along with fixing all the other tests?

lcreid replied:
Yes, you can just make the change on your same local branch as this PR originally came from. When you push you local branch to GitHub, GitHub will automatically update the PR.

Thanks again for the great work.

wpolicarpo commented:
@daveomcd do you mind rebasing your branch and resolve the conflicts? I merged #718, which contained must of the fixes in here.

daveomcd commented:
> @daveomcd do you mind rebasing your branch and resolve the conflicts? I merged #718, which contained must of the fixes in here.

My apologies on being a complete newb with Git.  How can I retrieve the conflicts I see below to resolve and push back up ?

lcreid commented:
@daveomcd you need to:

1. Update your repo from the original `rails-sqlserver` repo. This is typically called the upstream repo. https://www.atlassian.com/git/tutorials/git-forks-and-upstreams has a pretty good description of how to do this
2. Rebase or merge your `correcting-tests` branch with `upstream/master` (assuming you called the `rails-sqlserver` repo `upstream`). The link above describes how to update your `master` branch. To update your `correcting-tests` branch, just do the same thing, but in the `correcting-tests` branch
3. Resolve the merge conflicts. This typically means going through each file that has merge conflicts in your editor or IDE and manually correcting the files to what they should be. Hopefully in this case that should be obvious
4. Commit all the files
5. Push your `correcting-tests` branch to your GitHub repo, which will automatically update your PR

I hope this helps, but if not just keep asking questions. We're happy to help.

wpolicarpo commented:
Hey @daveomcd can I ask you to rebase this and resolve the conflicts so I can merge?

daveomcd commented:
> Hey @daveomcd can I ask you to rebase this and resolve the conflicts so I can merge?

My apologies I'll try working on it now. Things have been crazy! :(

daveomcd commented:
> Hey @daveomcd can I ask you to rebase this and resolve the conflicts so I can merge?

So... followed through everything and now I'm getting this message.

> Applying: corrected syntax for DEPRECATED warnings in test/cases
> No changes - did you forget to use 'git add'?
> If there is nothing left to stage, chances are that something else
> already introduced the same changes; you might want to skip this patch.
> 
> Resolve all conflicts manually, mark them as resolved with
> "git add/rm <conflicted_files>", then run "git rebase --continue".
> You can instead skip this commit: run "git rebase --skip".
> To abort and get back to the state before "git rebase", run "git rebase --abort".

When I run git rebase --continue.  Does this just mean I took so long that someone else already incorporated these changes? Ha.  If that's the case my sincerest apologies!

wpolicarpo commented:
@daveomcd don't be sorry. You're helping and we need to thank you for that.

First thing to do is to update your `upstream`  `master` branch as explained in https://github.com/rails-sqlserver/activerecord-sqlserver-adapter/pull/725#issuecomment-593759907.

Some changes in your PR were already merged into master indeed, but there are other we can still integrate. So you need to `git rebase -i master` and resolve the conflicts as they appear.

For each step after resolving the conflicts you will need to add your changed back `git add file_1 file_2`  and finally `git rebase --continue` to end up with a clean rebased tree.

If you prefer, you can `merge` instead of rebasing if it is easier to understand for now.

daveomcd commented:
> @daveomcd don't be sorry. You're helping and we need to thank you for that.
> 
> First thing to do is to update your `upstream` `master` branch as explained in [#725 (comment)](https://github.com/rails-sqlserver/activerecord-sqlserver-adapter/pull/725#issuecomment-593759907).
> 
> Some changes in your PR were already merged into master indeed, but there are other we can still integrate. So you need to `git rebase -i master` and resolve the conflicts as they appear.
> 
> For each step after resolving the conflicts you will need to add your changed back `git add file_1 file_2` and finally `git rebase --continue` to end up with a clean rebased tree.
> 
> If you prefer, you can `merge` instead of rebasing if it is easier to understand for now.

So I believe I did many of those steps.

```
git remote add upstream https://github.com/rails-sqlserver/activerecord-sqlserver-adapter.git
git fetch upstream
git checkout correcting-tests
git rebase upstream/master
```

Which gave me the following...

```
First, rewinding head to replay your work on top of it...
Applying: corrected syntax for DEPRECATED warnings in test/cases
Using index info to reconstruct a base tree...
M       test/cases/adapter_test_sqlserver.rb
M       test/cases/change_column_null_test_sqlserver.rb
M       test/cases/coerced_tests.rb
M       test/cases/column_test_sqlserver.rb
M       test/cases/connection_test_sqlserver.rb
M       test/cases/fetch_test_sqlserver.rb
M       test/cases/json_test_sqlserver.rb
M       test/cases/migration_test_sqlserver.rb
M       test/cases/pessimistic_locking_test_sqlserver.rb
M       test/cases/rake_test_sqlserver.rb
M       test/cases/schema_dumper_test_sqlserver.rb
M       test/cases/schema_test_sqlserver.rb
M       test/cases/showplan_test_sqlserver.rb
M       test/cases/specific_schema_test_sqlserver.rb
M       test/cases/transaction_test_sqlserver.rb
M       test/cases/trigger_test_sqlserver.rb
M       test/cases/utils_test_sqlserver.rb
M       test/cases/uuid_test_sqlserver.rb
Falling back to patching base and 3-way merge...
Auto-merging test/cases/transaction_test_sqlserver.rb
CONFLICT (content): Merge conflict in test/cases/transaction_test_sqlserver.rb
Auto-merging test/cases/specific_schema_test_sqlserver.rb
CONFLICT (content): Merge conflict in test/cases/specific_schema_test_sqlserver.rb
Auto-merging test/cases/schema_dumper_test_sqlserver.rb
Auto-merging test/cases/rake_test_sqlserver.rb
CONFLICT (content): Merge conflict in test/cases/rake_test_sqlserver.rb
Auto-merging test/cases/column_test_sqlserver.rb
CONFLICT (content): Merge conflict in test/cases/column_test_sqlserver.rb
Auto-merging test/cases/coerced_tests.rb
error: Failed to merge in the changes.
Patch failed at 0001 corrected syntax for DEPRECATED warnings in test/cases
Use 'git am --show-current-patch' to see the failed patch

Resolve all conflicts manually, mark them as resolved with
"git add/rm <conflicted_files>", then run "git rebase --continue".
You can instead skip this commit: run "git rebase --skip".
To abort and get back to the state before "git rebase", run "git rebase --abort".
```

I corrected them all then did...

```
git add .
git commit
git rebase --continue
```

Which resulted in...

```
Applying: corrected syntax for DEPRECATED warnings in test/cases
No changes - did you forget to use 'git add'?
If there is nothing left to stage, chances are that something else
already introduced the same changes; you might want to skip this patch.

Resolve all conflicts manually, mark them as resolved with
"git add/rm <conflicted_files>", then run "git rebase --continue".
You can instead skip this commit: run "git rebase --skip".
To abort and get back to the state before "git rebase", run "git rebase --abort".
```

tbonz commented:
@daveomcd it looks like you added a **git commit** in there instead of just continuing with the rebase after you added your corrections.  You only needed to:
```
git add .
git rebase --continue
```
You might reference this issue to see how you can reset your branch and allow you to continue with the rebase. https://stackoverflow.com/a/12163247/1245332

daveomcd commented:
> @daveomcd it looks like you added a **git commit** in there instead of just continuing with the rebase after you added your corrections. You only needed to:
> 
> ```
> git add .
> git rebase --continue
> ```
> 
> You might reference this issue to see how you can reset your branch and allow you to continue with the rebase. https://stackoverflow.com/a/12163247/1245332

@tbonz thanks for the help.  So I'm trying to get this going in between work.  I attempted to resolve the issue by doing the following but still seeing the same results.

```
git reset --soft HEAD^
git rebase --continue

```
```
Applying: corrected syntax for DEPRECATED warnings in test/cases
Applying: corrected syntax for DEPRECATED warnings in test/cases
Using index info to reconstruct a base tree...
M       Gemfile
M       test/cases/adapter_test_sqlserver.rb
M       test/cases/change_column_null_test_sqlserver.rb
M       test/cases/coerced_tests.rb
M       test/cases/column_test_sqlserver.rb
M       test/cases/connection_test_sqlserver.rb
M       test/cases/fetch_test_sqlserver.rb
M       test/cases/json_test_sqlserver.rb
M       test/cases/migration_test_sqlserver.rb
M       test/cases/pessimistic_locking_test_sqlserver.rb
M       test/cases/rake_test_sqlserver.rb
M       test/cases/schema_dumper_test_sqlserver.rb
M       test/cases/schema_test_sqlserver.rb
M       test/cases/showplan_test_sqlserver.rb
M       test/cases/specific_schema_test_sqlserver.rb
M       test/cases/transaction_test_sqlserver.rb
M       test/cases/trigger_test_sqlserver.rb
M       test/cases/utils_test_sqlserver.rb
M       test/cases/uuid_test_sqlserver.rb
Falling back to patching base and 3-way merge...
Auto-merging test/cases/transaction_test_sqlserver.rb
CONFLICT (content): Merge conflict in test/cases/transaction_test_sqlserver.rb
Auto-merging test/cases/specific_schema_test_sqlserver.rb
CONFLICT (content): Merge conflict in test/cases/specific_schema_test_sqlserver.rb
Auto-merging test/cases/rake_test_sqlserver.rb
CONFLICT (content): Merge conflict in test/cases/rake_test_sqlserver.rb
Auto-merging test/cases/column_test_sqlserver.rb
CONFLICT (content): Merge conflict in test/cases/column_test_sqlserver.rb
error: Failed to merge in the changes.
Patch failed at 0002 corrected syntax for DEPRECATED warnings in test/cases
Use 'git am --show-current-patch' to see the failed patch

Resolve all conflicts manually, mark them as resolved with
"git add/rm <conflicted_files>", then run "git rebase --continue".
You can instead skip this commit: run "git rebase --skip".
To abort and get back to the state before "git rebase", run "git rebase --abort".
```
At this point I went and corrected the conflicts and then proceeded with the following.

`git status`

```
rebase in progress; onto 20b0f60
You are currently rebasing branch 'correcting-tests' on '20b0f60'.
  (fix conflicts and then run "git rebase --continue")
  (use "git rebase --skip" to skip this patch)
  (use "git rebase --abort" to check out the original branch)

Unmerged paths:
  (use "git reset HEAD <file>..." to unstage)
  (use "git add <file>..." to mark resolution)

        both modified:   test/cases/column_test_sqlserver.rb
        both modified:   test/cases/rake_test_sqlserver.rb
        both modified:   test/cases/specific_schema_test_sqlserver.rb
        both modified:   test/cases/transaction_test_sqlserver.rb

no changes added to commit (use "git add" and/or "git commit -a")
```

```
git add .
git rebase --continue
```

Then seeing the same no changes message below.

```
Applying: corrected syntax for DEPRECATED warnings in test/cases
No changes - did you forget to use 'git add'?
If there is nothing left to stage, chances are that something else
already introduced the same changes; you might want to skip this patch.

Resolve all conflicts manually, mark them as resolved with
"git add/rm <conflicted_files>", then run "git rebase --continue".
You can instead skip this commit: run "git rebase --skip".
To abort and get back to the state before "git rebase", run "git rebase --abort".
```

I promise once I've graduated to the point past absolute newb at git I'll be able to contribute to more issues here without so many problems!

tbonz commented:
Can you try adding the files individually after saving them?  It seems like the files are not getting staged before you continue.  Maybe you try:
```
git add test/cases/column_test_sqlserver.rb
git status
```
and confirm the file is getting staged before you **git rebase --continue**

daveomcd commented:
> @daveomcd Can you try adding the files individually after saving them? It seems like the files are not getting staged before you continue. Maybe you try:
> 
> ```
> git add test/cases/column_test_sqlserver.rb
> git status
> ```
> 
> and confirm the file is getting staged before you **git rebase --continue**

Well the files that I need to resolve conflicts on I'm taking the "HEAD" changes each time.  So that makes me think the message is correct, that there doesn't appear to be any changes.  The conflicts were spots where they master uses paranthesis and I don't for my method arguments, etc.

tbonz commented:
@daveomcd I was able to clone down your fork and successfully rebase onto the upstream master.  I might recommend cloning your project down to a new directory and follow the steps again to rebase.  Here's what I did..

```
tbonz:~/dev
$ git clone git@github.com:daveomcd/activerecord-sqlserver-adapter.git
tbonz:~/dev
$ cd activerecord-sqlserver-adapter/
tbonz:~/dev/activerecord-sqlserver-adapter git:(master)
$ git remote add upstream git@github.com:rails-sqlserver/activerecord-sqlserver-adapter.git
tbonz:~/dev/activerecord-sqlserver-adapter git:(master)
$ git fetch upstream
tbonz:~/dev/activerecord-sqlserver-adapter git:(master)
$ git checkout correcting-tests
tbonz:~/dev/activerecord-sqlserver-adapter git:(𝘮 ← 2 correcting-tests)
$ git rebase upstream/master

Open editor and fix any conflicts (accept HEAD  changes as you mentioned) in each file, and save!

tbonz:~/dev/activerecord-sqlserver-adapter git:(upstream ⚡ detached@b833393) 3M 4B
$ git add test/cases/column_test_sqlserver.rb
tbonz:~/dev/activerecord-sqlserver-adapter git:(upstream ⚡ detached@b833393) 3M 3B
$ git add test/cases/rake_test_sqlserver.rb
tbonz:~/dev/activerecord-sqlserver-adapter git:(upstream ⚡ detached@b833393) 3M 2B
$ git add test/cases/specific_schema_test_sqlserver.rb
tbonz:~/dev/activerecord-sqlserver-adapter git:(upstream ⚡ detached@b833393) 3M 1B
$ git add test/cases/transaction_test_sqlserver.rb
tbonz:~/dev/activerecord-sqlserver-adapter git:(upstream ⚡ detached@b833393) 3M
$ git status
rebase in progress; onto b833393
You are currently rebasing branch 'correcting-tests' on 'b833393'.
  (all conflicts fixed: run "git rebase --continue")

Changes to be committed:
  (use "git reset HEAD <file>..." to unstage)

	modified:   Gemfile
	modified:   test/cases/coerced_tests.rb
	modified:   test/cases/schema_dumper_test_sqlserver.rb

tbonz:~/dev/activerecord-sqlserver-adapter git:(upstream ⚡ detached@b833393) 3M
$ git rebase --continue
Applying: corrected syntax for DEPRECATED warnings in test/cases
Applying: corrected syntax for DEPRECATED warnings in test/cases
```

The next step would be to force push this back to origin.

```
tbonz:~/dev/activerecord-sqlserver-adapter git:(𝘮 ← 2 correcting-tests 2⇵20)
$ git push -f origin correcting-tests
```

Let me know if this helps, or I could offer a meetup on google to assist any further.

daveomcd commented:
> @daveomcd I was able to clone down your fork and successfully rebase onto the upstream master. I might recommend cloning your project down to a new directory and follow the steps again to rebase. Here's what I did..
> 
> ```
> tbonz:~/dev
> $ git clone git@github.com:daveomcd/activerecord-sqlserver-adapter.git
> tbonz:~/dev
> $ cd activerecord-sqlserver-adapter/
> tbonz:~/dev/activerecord-sqlserver-adapter git:(master)
> $ git remote add upstream git@github.com:rails-sqlserver/activerecord-sqlserver-adapter.git
> tbonz:~/dev/activerecord-sqlserver-adapter git:(master)
> $ git fetch upstream
> tbonz:~/dev/activerecord-sqlserver-adapter git:(master)
> $ git checkout correcting-tests
> tbonz:~/dev/activerecord-sqlserver-adapter git:(𝘮 ← 2 correcting-tests)
> $ git rebase upstream/master
> tbonz:~/dev/activerecord-sqlserver-adapter git:(upstream ⚡ detached@b833393) 3M 4B
> $ git status
> rebase in progress; onto b833393
> You are currently rebasing branch 'correcting-tests' on 'b833393'.
>   (fix conflicts and then run "git rebase --continue")
>   (use "git rebase --skip" to skip this patch)
>   (use "git rebase --abort" to check out the original branch)
> 
> Changes to be committed:
>   (use "git reset HEAD <file>..." to unstage)
> 
> 	modified:   Gemfile
> 	modified:   test/cases/coerced_tests.rb
> 	modified:   test/cases/schema_dumper_test_sqlserver.rb
> 
> Unmerged paths:
>   (use "git reset HEAD <file>..." to unstage)
>   (use "git add <file>..." to mark resolution)
> 
> 	both modified:   test/cases/column_test_sqlserver.rb
> 	both modified:   test/cases/rake_test_sqlserver.rb
> 	both modified:   test/cases/specific_schema_test_sqlserver.rb
> 	both modified:   test/cases/transaction_test_sqlserver.rb
> ```
> 
> Open editor and fix any conflicts (accept HEAD changes as you mentioned) in each file, and save!
> 
> ```
> tbonz:~/dev/activerecord-sqlserver-adapter git:(upstream ⚡ detached@b833393) 3M 4B
> $ git add test/cases/column_test_sqlserver.rb
> tbonz:~/dev/activerecord-sqlserver-adapter git:(upstream ⚡ detached@b833393) 3M 3B
> $ git add test/cases/rake_test_sqlserver.rb
> tbonz:~/dev/activerecord-sqlserver-adapter git:(upstream ⚡ detached@b833393) 3M 2B
> $ git add test/cases/specific_schema_test_sqlserver.rb
> tbonz:~/dev/activerecord-sqlserver-adapter git:(upstream ⚡ detached@b833393) 3M 1B
> $ git add test/cases/transaction_test_sqlserver.rb
> tbonz:~/dev/activerecord-sqlserver-adapter git:(upstream ⚡ detached@b833393) 3M
> $ git status
> rebase in progress; onto b833393
> You are currently rebasing branch 'correcting-tests' on 'b833393'.
>   (all conflicts fixed: run "git rebase --continue")
> 
> Changes to be committed:
>   (use "git reset HEAD <file>..." to unstage)
> 
> 	modified:   Gemfile
> 	modified:   test/cases/coerced_tests.rb
> 	modified:   test/cases/schema_dumper_test_sqlserver.rb
> 
> tbonz:~/dev/activerecord-sqlserver-adapter git:(upstream ⚡ detached@b833393) 3M
> $ git rebase --continue
> Applying: corrected syntax for DEPRECATED warnings in test/cases
> Applying: corrected syntax for DEPRECATED warnings in test/cases
> tbonz:~/dev/activerecord-sqlserver-adapter git:(𝘮 ← 2 correcting-tests 2⇵20)
> $ git status
> On branch correcting-tests
> Your branch and 'origin/correcting-tests' have diverged,
> and have 20 and 2 different commits each, respectively.
>   (use "git pull" to merge the remote branch into yours)
> 
> nothing to commit, working tree clean
> ```
> 
> The next step would be to force push this back to origin.
> 
> ```
> tbonz:~/dev/activerecord-sqlserver-adapter git:(𝘮 ← 2 correcting-tests 2⇵20)
> $ git push -f origin correcting-tests
> ```
> 
> Let me know if this helps, or I could offer a meetup on google to assist any further.

@tbonz oh man you're the best. Trying it now. Thanks so much for taking the time to teach me how to do this.

daveomcd commented:
@tbonz thanks! I think that worked.  Only thing I noticed was I could probably consolidate my commits in my log.  But I didn't due to I was nervous I'd screw it up again!  But I can't do that next if necessary.  @wpolicarpo thank you as well for helping me out.  I look forward to trying to help out to another area and getting this process down.
